scripts/jenkins/cloud/ansible/roles/heat-generator/templates/heat-template.yaml

```diff
@@ -71,8 +72,22 @@ resources:
       network_id: { get_resource: {{ name }}_network }
 {%   if network.cidr is defined %}
       cidr: "{{ network.cidr }}"
+{%       if ( network.cidr | ipv6 ) %}
+      ip_version: 6
+{%          if network.is_conf or network.is_mgmt %}
+      ipv6_ra_mode: dhcpv6-stateful
+      ipv6_address_mode: dhcpv6-stateful
+{%          endif %}
+{%       else %}
+      ip_version: 4
+{%       endif %}
 {%   elif network.allocation_pools is not defined %}
       cidr: { get_param: default_subnet_cidr }
+      ip_version: {{ ipv6 | ternary(6, 4) }}
```

Comment:
here can deduce the ip version the same way it is deduced aove in line 50. we live it to the user choice (ipv6 = true) since that will determine what version teh default_subnet_cidr is